relayer/src/foreign_client.rs

```diff
@@ -778,6 +778,29 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
     }
 
     pub fn refresh(&mut self) -> Result<Option<Vec<IbcEvent>>, ForeignClientError> {
+        match self.try_refresh() {
+            Ok(res) => {
+                if let Some(ibc_events) = res.clone() {
+                    match ibc_events
+                        .into_iter()
+                        .find(|e| e.event_type() == IbcEventType::ChainError)
+                    {
+                        Some(ev) => {
+                            return Err(ForeignClientError::chain_error_event(
+                                self.dst_chain().id(),
+                                ev,
+                            ));
+                        }
```

Comment:
There are many different execution branches and it's not clear what are the post-conditions of the `refresh` method. Can we briefly describe that?

Let's maybe also put a comment to describe what is this match arm for. There is also an `Ok(res)` and a `Err(e` branches.